### PR TITLE
Improve eventing example

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- change runtime to nodejs16
- fix git reference to main
- improve async call to eventing proxy
- wrap the call with custom span with tags

### CODE REVIEW

**Comment by pPrecel** [comment]
This example is not working for me 😢 I've tested it with a such spell:

```bash
 curl -ik "https://incoming.<DOMAIN>/"

HTTP/2 200
x-powered-by: Express
access-control-allow-origin: *
content-type: application/json; charset=utf-8
content-length: 179
etag: ...
date: ...
x-envoy-upstream-service-time: 19
server: istio-envoy

{"type":"sap.kyma.custom.acme.payload.sanitised.v1","source":"kyma","specversion":"1.0","id":"a7880351-7c28-4c87-9240-b0e908f60c98","data":{},"datacontenttype":"application/json"}
```

Looks ok, but received did not receive any message and the emitter logs an error. Logs from the emitter:

```bash
kubectl logs event-emitter-rd8p2-596cc544d4-27gzd -f

> nodejs16-runtime@0.1.0 start
> node server.js

user code loaded in 0sec 27.214074ms
sanitising data...
{}
Payload pushed to sap.kyma.custom.acme.payload.sanitised.v1 {
  type: 'sap.kyma.custom.acme.payload.sanitised.v1',
  source: 'kyma',
  eventtypeversion: undefined,
  specversion: '1.0',
  id: '2766b3bc-c3c3-4003-bfc9-15bd8bdddcf4',
  data: {},
  datacontenttype: 'application/json'
}
Caught exception: TypeError [ERR_HTTP_INVALID_HEADER_VALUE]: Invalid value "undefined" for header "ce-eventtypeversion"
```

Any idea what is going on?

**Comment by pPrecel** [comment]
I see in [the Readme file](https://github.com/kwiatekus/examples/blob/improve-eventing-example/incluster_eventing/README.md#deploy-via-kubectl) info about the `k8s-resources` folder. Shall we render the changes to this folder? or maybe just remove it? right now there are yaml's which are not up-to-date.

**Comment by pPrecel** [comment]
Last thing. I found info that users can run this example in the docker using `kyma run function` command [here](https://github.com/kwiatekus/examples/blob/improve-eventing-example/incluster_eventing/README.md#deploy-via-kyma-cli). Are you sure it's still actual?

**Comment by pPrecel** [comment]
> > Last thing. I found info that users can run this example in the docker using `kyma run function` command [here](https://github.com/kwiatekus/examples/blob/improve-eventing-example/incluster_eventing/README.md#deploy-via-kyma-cli). Are you sure it's still actual?
> 
> looks like its actual. I can run it locally via kyma CLI

Yes, but this is an example of incluster-eventing :D How would it be working using the `kyma run function` command?

**Review by pPrecel — APPROVED** [review]
LGTM

File: incluster_eventing/README.md [hunk 1]
```diff
@@ -16,23 +16,39 @@ This example also provides a template for a git project with Kyma Functions. Ple
 ## Deploy
 ### Deploy via Kyma CLI
 
-Use your favorite IDE to change the logic in the handler.js files.
-Use Kyma CLI from the Function directory (for example, `incluster_eventing/src/emitter-fn`) to:
- - Run the Function locally (using `kyma run function`) 
- - Deploy the Function to Kyma runtime (using `kyma apply function` )
+You can deploy each  Function separately using Kyma CLI by running `kyma apply function` in each of the functions source folders.
```
Comment (grego952) [inline incluster_eventing/README.md:19]:
```suggestion
You can deploy each Function separately using Kyma CLI by running `kyma apply function` in each of the Function's source folders.
```

File: incluster_eventing/README.md [hunk 2]
```diff
@@ -16,23 +16,39 @@ This example also provides a template for a git project with Kyma Functions. Ple
...
 
 You can find all installation steps in the [Set asynchronous communication between Functions](https://kyma-project.io/docs/kyma/latest/03-tutorials/00-serverless/svls-11-set-asynchronous-connection-of-functions/) tutorial.
 
+### Deploy via kubectl
+
+Deploy to Kyma runtime manually using `kubectl apply`  or `make deploy` target.
```
Comment (grego952) [inline incluster_eventing/README.md:25]:
```suggestion
Deploy to Kyma runtime manually using `kubectl apply` or `make deploy` target.
```

File: incluster_eventing/README.md [hunk 3]
```diff
@@ -16,23 +16,39 @@ This example also provides a template for a git project with Kyma Functions. Ple
...
 
-### Deploy via kubectl
 
-Render Kubernetes manifests using the `make render` target. This outputs Kubernetes manifests to the `k8s-resources` folder.
 
-Deploy to Kyma runtime manually using kubectl or `make deploy`.
-There is also a [github workflow](.github/workflows/deploy.yml) included which you can use as a template to come up with own automated CI/CD.
+### Test the application
+
+Send an HTTP request to the emitter function
```
Comment (grego952) [inline incluster_eventing/README.md:36]:
```suggestion
Send an HTTP request to the emitter Function
```

File: incluster_eventing/README.md [hunk 4]
```diff
@@ -16,23 +16,39 @@ This example also provides a template for a git project with Kyma Functions. Ple
...
+
+```bash
+curl -H "Content-Type: application/cloudevents+json" -X POST  https://incoming.{your cluster domain} -d '{"foo":"bar"}'
+Event sent%
+```
+
+Fetch the logs of the receiver function to observe the message arrived.
```
Comment (grego952) [inline incluster_eventing/README.md:43]:
```suggestion
Fetch the logs of the receiver Function to observe the incoming message.
```

**Review by pPrecel — APPROVED** [review]
Now it looks even better

**Review by grego952 — APPROVED** [review]

**Pull request merged**